### PR TITLE
MappingError: expose the first error in Exception message

### PR DESCRIPTION
**Pull request by Slamdunk**
Hi, the [message customization](https://github.com/CuyZ/Valinor#message-customization) is great, but a first feedback without it would be very useful to speed up debugging.

Before and after this PR:

```diff
-CuyZ\Valinor\Mapper\MappingError: Could not map type `array{foo: string, bar: int}` with the given source.
+CuyZ\Valinor\Mapper\MappingError: Could not map type `array{foo: string, bar: int}` with the given source. First error is:
+Path: foo
+Given value: array (empty)
+Error: Cannot be empty and must be filled with a value matching type `string`.
```

### CODE REVIEW

**Comment by romm** [comment]
Hi @Slamdunk, that was a very good idea. I took some time to think about it and reworked a bit around your initial contribution.

Hope you'll like it!

File: tests/Integration/Mapping/MappingErrorTest.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+final class MappingErrorTest extends IntegrationTest
+{
+    public function test_mapping_error_code_is_correct(): void
+    {
+        try {
+            (new MapperBuilder())->mapper()->map('string', ['foo']);
+
+            self::fail();
+        } catch (MappingError $exception) {
+            self::assertSame(1617193185, $exception->getCode());
```
Comment (Slamdunk) [inline tests/Integration/Mapping/MappingErrorTest.php:20]: `try-catch` statements inside tests may be misleading: did the author want to test the Exception, or just protect the test from and edge case?
This one may be easy to read, but in the long run it becomes harder: I suggest to leverage the built-in expectations next time: https://phpunit.readthedocs.io/en/9.5/writing-tests-for-phpunit.html#testing-exceptions

**Review by Slamdunk — COMMENTED** [review]
Thank you @romm, awesome improvements over my first implementation :muscle:

File: tests/Integration/Mapping/MappingErrorTest.php (hunk 1 above)
Comment (romm) [reply tests/Integration/Mapping/MappingErrorTest.php:20]: True, I actually wonder why I didn't go this way. Gonna make the change now!
